playbooks/cloud-provider-openstack-acceptance-test-e2e-conformance/run.yaml

```diff
@@ -1,157 +1,146 @@
-          '{{ kubectl }}' create clusterrolebinding --user system:serviceaccount:kube-system:cloud-node-controller kube-system-cluster-admin-3 --clusterrole cluster-admin
-          '{{ kubectl }}' create clusterrolebinding --user system:serviceaccount:kube-system:cloud-controller-manager kube-system-cluster-admin-4 --clusterrole cluster-admin
-          '{{ kubectl }}' create clusterrolebinding --user system:serviceaccount:kube-system:shared-informers kube-system-cluster-admin-5 --clusterrole cluster-admin
-          '{{ kubectl }}' create clusterrolebinding --user system:kube-controller-manager  kube-system-cluster-admin-6 --clusterrole cluster-admin
-
-          # Sleep to wait for creating serviceaccount default/default complete
-          sleep 5
-
-          # Set admin kubeconfig for running e2e tests
-          export KUBECONFIG=/var/run/kubernetes/admin.kubeconfig
-
-          # Build kubetest
-          # go get kubetest is failing from k8s or cpo directory
-          cd ..
-          export GO111MODULE=on
-          # export GOPROXY=https://proxy.golang.org
-          # export GOSUMDB=sum.golang.org
-
-          git clone https://github.com/kubernetes/test-infra
-          cd test-infra
-          GOBIN='{{ k8s_src_dir }}' go install ./kubetest
+          # Create cloud-config
+          kubectl -n kube-system create secret generic cloud-config --from-file=cloud.conf=/etc/kubernetes/cloud-config -oyaml
 
-          # Build all binaries
-          export KUBE_FASTBUILD=true
-          export KUBERNETES_CONFORMANCE_TEST=y
-          # Go modules not supported by kubernetes before v1.15
-          if [ '{{ k8s_version }}' == 'release-1.14' ]; then
-            unset GO111MODULE
-            unset GOSUMDB
-            unset GOPROXY
-          fi
+    - name: depoy openstack cloud controller manager
+      shell:
+        chdir: '{{ k8s_os_provider_src_dir }}'
+        executable: /bin/bash
+        cmd: |
+          set -ex
 
-          # TEMP - Dump Memory Every Second
-          free -lmthw -c10000 >> $LOG_DIR/system_memory.log &
+          kubectl apply -f cluster/addons/rbac/
+          kubectl apply -f manifests/controller-manager/openstack-cloud-controller-manager-ds.yaml
 
-          # Run e2e test using local deployment/provider
-          # Must run kubetest under kubernetes root directory
-          cd '{{ k8s_src_dir }}'
+    - name: get kubeadm join command
+      shell: kubeadm token create --print-join-command 2>/dev/null
+      register: kubeadm_join_cmd
 
-          # temproary change to solve memory issue
-          make WHAT="test/e2e/e2e.test vendor/github.com/onsi/ginkgo/ginkgo cmd/kubectl"
+- hosts: k8s-node
+  name: join kubernetes nodes
+  become: yes
+  roles:
+    - role: create-multinodes-k8s-cluster-with-kubeadm
+      k8s_role_to_deploy:
+        - node
+      kubeadm_join_cmd: "{{ hostvars['k8s-master']['kubeadm_join_cmd']['stdout'] }}"
```

Comment:
Don't we need to wait for complete node joining?
Basically it takes time to become "Ready" state on worker node after operating `kubeadm join` command.

Reply:
Done